bluemira/equilibria/opt_constraints.py

```diff
@@ -20,262 +20,848 @@
-    eq: Equilibrium
-        Equilibrium object with which to calculate constraints
-    n_PF: int
-        Number of PF coils
-    n_CS: int
-        Number of CS coils
+    coilset: CoilSet
+        Coilset for which to constrain the fields in the coils
     PF_Fz_max: float
-        Maximum vertical force on each PF coil [MN]
+        Maximum absolute vertical force in a PF coil [MN]
     CS_Fz_sum_max: float
-        Maximum total vertical force on the CS stack [MN]
+        Maximum absolute vertical force sum in the CS stack [MN]
     CS_Fz_sep_max: float
-        Maximum vertical separation force in the CS stack [MN]
-    scale: float
-        Current scale with which to calculate the constraints
-
-    Returns
-    -------
-    constraint: np.ndarray
-        Updated constraint vector
+        Maximum separation vertical force between two CS modules [MN]
+    tolerance: Union[float, np.ndarray]
+        Tolerance with which the inequality constraints will be met
 
     Notes
     -----
     TODO: Presently only handles CoilSets with Coils (SymmetricCircuits not yet
     supported)
     """
-    # get coil force and jacobian
-    F, dF = eq.force_field.calc_force(vector * scale)  # noqa :N803
-    F /= scale  # Scale down to MN
-    # dF /= self.scale
-
-    # Absolute vertical force constraint on PF coils
-    constraint[:n_PF] = F[:n_PF, 1] ** 2 - PF_Fz_max**2
-
-    if n_CS != 0:
-        # vertical forces on CS coils
-        cs_fz = F[n_PF:, 1]
-        # vertical force on CS stack
-        cs_z_sum = np.sum(cs_fz)
-        # Absolute sum of vertical force constraint on entire CS stack
-        constraint[n_PF] = cs_z_sum**2 - CS_Fz_sum_max**2
-        for i in range(n_CS - 1):  # evaluate each gap in CS stack
-            # CS seperation constraints
-            f_sep = np.sum(cs_fz[: i + 1]) - np.sum(cs_fz[i + 1 :])
-            constraint[n_PF + 1 + i] = f_sep - CS_Fz_sep_max
-
-    # calculate constraint jacobian
-    if grad.size > 0:
-        # Absolute vertical force constraint on PF coils
-        grad[:n_PF] = 2 * dF[:n_PF, :, 1]
-
-        if n_CS != 0:
-            # Absolute sum of vertical force constraint on entire CS stack
-            grad[n_PF] = 2 * np.sum(dF[n_PF:, :, 1], axis=0)
-
-            for i in range(n_CS - 1):  # evaluate each gap in CS stack
-                # CS separation constraint Jacobians
-                f_up = np.sum(dF[n_PF : n_PF + i + 1, :, 1], axis=0)
-                f_down = np.sum(dF[n_PF + i + 1 :, :, 1], axis=0)
-                grad[n_PF + 1 + i] = f_up - f_down
-    return constraint
-
-
-def coil_field_constraints(constraint, vector, grad, eq, B_max, scale):
+
+    def __init__(
+        self, coilset, PF_Fz_max, CS_Fz_sum_max, CS_Fz_sep_max, tolerance=1.0e-6
+    ):
+
+        n_PF = coilset.n_PF
+        n_CS = coilset.n_CS
+        if n_CS == 0:
+            n_f_constraints = n_PF
+        else:
+            n_f_constraints = n_PF + n_CS
```

Comment:
No need for the distinction I dont think

```suggestion
        n_f_constraints = n_PF + n_CS
```